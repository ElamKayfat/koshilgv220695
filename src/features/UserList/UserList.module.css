.root {
  max-width: 400px;
  max-height: 500px;
  margin: auto;
  margin-top: 50px;
}