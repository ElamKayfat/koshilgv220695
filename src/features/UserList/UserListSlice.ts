import { createSlice } from "@reduxjs/toolkit";

export type UserDataType = {
  phone: string;
  name: string;
  nickname: string;
  email: string;
  position: string;
  photo: string;
};

type UserListState = {
  userListData: UserDataType[];
}

const initialState = {
  userListData: []
} as UserListState;

const userListSlice = createSlice({
  name: "userList",
  initialState,
  reducers: {
    setUserListData(state, action) {
      state.userListData = action.payload.userListData;
    },
  },
});

export const { setUserListData } = userListSlice.actions;
export default userListSlice.reducer;
