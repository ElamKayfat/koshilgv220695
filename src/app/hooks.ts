import { useEffect } from 'react';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import { setUserListData } from '../features/UserList/UserListSlice';
import { mockUserList } from './mockData';
import type { RootState, AppDispatch } from './store';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const useFetchUserListAndSave = () => {
  const dispatch = useDispatch()
  
  useEffect(() => {
    setTimeout(() => {
      dispatch(setUserListData({ userListData: mockUserList }))
    }, 200)
  }, [])
}
