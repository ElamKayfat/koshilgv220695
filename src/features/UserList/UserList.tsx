import UserListItem from "../UserListItem/UserListItem";
import { UserDataType } from "./UserListSlice";
import styles from './UserList.module.css'

type UserListProps = {
  userList: UserDataType[];
  onSelectUser: (userData: UserDataType) => void
};

function UserList(props: UserListProps) {
  return (
    <div className={styles.root}>
      {props.userList.map((userData) => (
        <UserListItem
          key={userData.email}
          userData={userData}
          onClick={props.onSelectUser}
        />
      ))}
    </div>
  );
}

export default UserList;
