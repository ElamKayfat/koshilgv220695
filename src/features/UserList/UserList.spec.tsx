import { fireEvent, getByTestId, getByText, render } from '@testing-library/react';
import { mockUserList } from '../../app/mockData';
import UserList from './UserList';

test('Make snapshot. Render with data', () => {
  const mockSlectUserFn = jest.fn()
  const { container } = render(<UserList userList={mockUserList}  onSelectUser={mockSlectUserFn}/>);

  expect(container).toMatchSnapshot();
});