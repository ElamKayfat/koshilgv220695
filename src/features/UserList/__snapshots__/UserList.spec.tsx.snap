// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make snapshot. Render with data 1`] = `
<div>
  <div
    class="root"
  >
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Geordan Aaryn"
          class="avatar"
          src="1.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Geordan Aaryn
          </span>
          <span
            class="nickName"
          >
            @geordanaaryn
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Geordan Aaryn view button"
      >
        View
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Brazil Izair"
          class="avatar"
          src="2.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Brazil Izair
          </span>
          <span
            class="nickName"
          >
            @brazilizair
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Brazil Izair view button"
      >
        View
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Selman Winston"
          class="avatar"
          src="3.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Selman Winston
          </span>
          <span
            class="nickName"
          >
            @selmanwinston
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Selman Winston view button"
      >
        View
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Kaine Dutch"
          class="avatar"
          src="4.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Kaine Dutch
          </span>
          <span
            class="nickName"
          >
            @kainedutch
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Kaine Dutch view button"
      >
        View
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Kerby Linton"
          class="avatar"
          src="5.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Kerby Linton
          </span>
          <span
            class="nickName"
          >
            @kerbylinton
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Kerby Linton view button"
      >
        View
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Gilad Delvon"
          class="avatar"
          src="6.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Gilad Delvon
          </span>
          <span
            class="nickName"
          >
            @giladdelvon
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Gilad Delvon view button"
      >
        View
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Augusto Castor"
          class="avatar"
          src="7.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Augusto Castor
          </span>
          <span
            class="nickName"
          >
            @augustocastor
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Augusto Castor view button"
      >
        View
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Zian Khyre"
          class="avatar"
          src="8.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Zian Khyre
          </span>
          <span
            class="nickName"
          >
            @ziankhyre
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Zian Khyre view button"
      >
        View
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Sheldon Neiko"
          class="avatar"
          src="9.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Sheldon Neiko
          </span>
          <span
            class="nickName"
          >
            @sheldonneiko
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Sheldon Neiko view button"
      >
        View
      </button>
    </div>
    <div
      class="root"
    >
      <div
        class="mainInfoContainer"
      >
        <img
          alt="Berenger Altay"
          class="avatar"
          src="10.jpg"
        />
        <div
          class="wrapper"
        >
          <span
            class="name"
          >
            Berenger Altay
          </span>
          <span
            class="nickName"
          >
            @berengeraltay
          </span>
        </div>
      </div>
      <button
        class="button"
        data-testid="Berenger Altay view button"
      >
        View
      </button>
    </div>
  </div>
</div>
`;
